app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java

```diff
@@ -49,6 +35,15 @@ public MyViewHolder onCreateViewHolder(final ViewGroup parent, final int viewTyp
     public void onBindViewHolder(final MyViewHolder holder, final int position) {
         final TextView categoryName = (TextView) holder.itemView.findViewById(R.id.categories_name);
         categoryName.setText(Category.values()[position].getName());
+        categoryName.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(view.getContext(), Category.values()[position].getName(), Toast.LENGTH_SHORT).show();
```

Comment:
inside this method you twice use Category.values()[position] - this is called code duplication. Extract this as local varibal final Category category =  Category.values()[position]